application/src/main/scala/com/azavea/franklin/extensions/validation/package.scala

```diff
@@ -1,50 +1,238 @@
+      m: Map[String, T => T]
+  ): F[ExtensionRef[F, T]] =
+    Ref.of(m)
+
+  private val knownItemExtensions: Map[String, StacItem => StacItem] = Map(
+    "label"                                                                                  -> labelItemLinkValidator,
+    "https://raw.githubusercontent.com/stac-extensions/label/v1.0.0/json-schema/schema.json" -> labelItemLinkValidator,
+    "eo"                                                                                     -> ((item: StacItem) => item.validate[EOItemExtension]("eo")),
+    "https://raw.githubusercontent.com/stac-extensions/eo/v1.0.0/json-schema/schema.json" -> (
+      (item: StacItem) => item.validate[EOItemExtension]("eo")
+    )
+  )
+
+  def itemExtensionsRef[F[_]: Sync]       = getExtensionsRef[F, StacItem](knownItemExtensions)
+  def collectionExtensionsRef[F[_]: Sync] = getExtensionsRef[F, StacCollection]
+
+  private def readAsSchema[F[_]: Sync](
+      extension: String
+  )(
+      implicit backend: SttpBackend[F, Nothing, NothingT]
+  ): F[Either[String, Schema]] = {
+    val extensionUriParsed = SttpUri.parse(extension)
+    extensionUriParsed flatTraverse { extensionUri =>
+      basicRequest.get(extensionUri).response(asJson[Json]).send[F].attempt map {
+        case Right(resp) =>
+          resp.body
+            .bimap(
+              {
+                case DeserializationError(_, _) =>
+                  s"Value at $extensionUri was not valid json and can't be used for extension schema validation"
+                case HttpError(_, code) =>
+                  s"Could not read from $extensionUri while attempting to validate extensions. Response code was: $code"
+                case _ =>
+                  s"Encountered an unknown error trying to read $extensionUri while validating extensions"
+              },
+              js => {
+                Try(Schema.load(js))
+              }
+            )
+            .flatMap(
+              Either
+                .fromTry(_)
+                .leftMap(_ => s"""| Failed to load a schema from the json at $extensionUri.
+                      | This can commonly be caused by links that can't be resolved during load.
+                      | Not validating the extension at $extensionUri.""".trim.stripMargin)
+            )
+        case Left(_) =>
+          Either.left[String, Schema](
+            s"Request for extension uri at $extension failed hard and we couldn't try to load a schema from its response"
+          )
       }
-    })
+    }
   }
 
-  def getLinkValidator(extensions: List[String]): StacLink => StacLink = {
-    makeValidator(extensions map { s => (link: StacLink) =>
-      ExtensionName.fromString(s) match {
-        case Label        => link.validateWhen[LabelLinkExtension]("label", _.rel == StacLinkType.Source)
-        case Layer        => link
-        case EO           => link
-        case Unchecked(_) => link
+  private val itemSchemaValidator = (schema: Schema) =>
```

Comment:
Should not this be just `def` ?

Reply:
When I'm explicitly constructing anonymous functions to fill type holes I frequently use `val` ("I need a value of type a -> b -> whatever" tricks my brain). What's the difference here though?

Reply:
No difference, just a question. :+1:

Reply:
Oh ok then the answer is "this thing my brain does when I'm passing functions around" 🙃

Reply:
Sounds good to me 🎸